### ISSUE TITLE
REQUEST: Ideas for how to replace IFTTT

### ISSUE THREAD

**Issue by Alfiegerner** [OmerTu/GoogleHomeKodi#319]
Now that IFTTT is moving behind a paywall, anybody have any ideas for a replacement service?

**Comment by keydon** [comment]
Oh no, that's one bad piece of news.
At least they leave three applets for free, which let's us continue working
with the broker-applet.

But I guess it's time to look deeper into googels dialogflow/action
builder. But I doubt we will be able to "release" an action to the public,
meaning you guys would have to register as developers with Google and then
import, whatever we come up with. It definitely does not make thinks
easier.

**Comment by Alfiegerner** [comment]
Yeah it's a real pain, particularly if there are other services you need the IFTTT slots for. Let me know if you want someone to help test.

**Comment by twochannel** [comment]
integrating with existing eco systems may be an option - e.g. home assistant.   Home assistant allows you to use google assistant a couple of ways.  Google's new local api sdk could be another option.

**Comment by talondnb** [comment]
Keen on this!

**Comment by apaperclip** [comment]
I dont think home assistant would work unless we can get the full voice-to-text string from google assistant into home assistant. 

Today home assistant integrates specific domains (switches, lights, hvac, etc) to equivalent google assistant traits. It's not open ended enough for how GHK uses  google assistant -> ifttt.

Home Assistant does have its own voice assistant but that would obviously not use google assistant.

**Comment by talondnb** [comment]
How about using something like this?

https://www.home-assistant.io/integrations/dialogflow/

**Comment by apaperclip** [comment]
> How about using something like this?
> 
> https://www.home-assistant.io/integrations/dialogflow/

I looked into it a little bit. Most things I saw said you have to "hey google, talk to <app name>" to initiate.  I'll take a closer look.

**Comment by Sk3yNet** [comment]
I like [Google Local fulfillment](https://developers.google.com/assistant/smarthome/concepts/local) as it eliminates IFTTT without adding a different replacement service. It can be [programmed in JavaScript](https://developers.google.com/assistant/smarthome/develop/local), can send http requests & has [example code](https://github.com/actions-on-google/smart-home-local).

Just my 2 cents from a grateful user of this project!

**Comment by keydon** [comment]
> I like [Google Local fulfillment](https://developers.google.com/assistant/smarthome/concepts/local) as it eliminates IFTTT without adding a different replacement service. It can be [programmed in JavaScript](https://developers.google.com/assistant/smarthome/develop/local), can send http requests & has [example code](https://github.com/actions-on-google/smart-home-local).

so far as I understand local fullfillment is meant as a addition. google still requires a central service it can call for fullfillment, but it may decide to switch to lokal fullfillment once the "device" is discovered. Not sure how big our userbase is. I might consider setting up a central service with limited capabilities (i.e. no youtube-api) and hoping for traffic beeing very low due to local fullfillment.

**Comment by outdoor720** [comment]
Has anyone tried: Tasker/Autovoice and Yatse?

https://forum.joaoapps.com/index.php?resources/control-kodi-the-easy-way-with-autovoice-and-yatse.317/

**Comment by the4anoni** [comment]
Any new on this? Parse broker doesn't works great, known commands like kodi stop doesn't work.

**Comment by DarthNucifera** [comment]
We really need this after the news from IFTTT today. At the end of the month this whole project will cease to function. No more text elements are allowed from IFTTT to Google Home.

**Comment by talondnb** [comment]
> We really need this after the news from IFTTT today. At the end of the month this whole project will cease to function. No more text elements are allowed from IFTTT to Google Home.

Do you have a link to the news?

**Comment by DarthNucifera** [comment]
https://9to5google.com/2022/08/02/google-assistant-ifttt/

**Comment by apaperclip** [comment]
I bet with the new voice assistant features in home assistant, this ability could be handled natively and locally with HA. (with some development like building the matching logic of spoken text to items in the kodi DB)

I do not know if google home could still be used though.

**Comment by apaperclip** [comment]
Looks like this feature gets us one step closer to an HA based solution
https://rc.home-assistant.io/blog/2023/07/26/release-20238/#wildcard-support-for-sentence-triggers

It may work well enough now but we'd be missing the fuzzy logic matching. That need to be a service call or custom voice assitant.
